File: CS/WindowsFormsApplication1/WizardControlAnimationHelper.cs
using DevExpress.Utils.Animation;
using DevExpress.XtraWizard;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace WindowsFormsApplication1 {

    public enum TransitionType { PushTransition, ShapeTransition, FadeTransition, ClockTransition, DissolveTransition, SlideFadeTransition, CoverTransition, CombTransition }

    public class WizardControlAnimationHelper : Component {
        private Transition transition;
        private WizardControl _WizardControl;
        private TransitionManager _TransitionManager;
        private TransitionType _TransitionType;
        


        public WizardControlAnimationHelper() {
            transition = new Transition();
            transition.ShowWaitingIndicator = DevExpress.Utils.DefaultBoolean.False;
            TransitionType = WindowsFormsApplication1.TransitionType.PushTransition;
        }

        ~WizardControlAnimationHelper() {
            if (TransitionManager != null) {
                TransitionManager.Transitions.Remove(transition);
            }
        }

        public TransitionType TransitionType {
            get { return _TransitionType; }
            set {
                if (_TransitionType != value) {
                    _TransitionType = value;
                    if (!(this.DesignMode)) {
                        CustomizeTransitionType();
                    }
                }
            }
        }

        public WizardControl WizardControl {
            get {
                return _WizardControl;
            }
            set {
                if (_WizardControl != value) {
                    if (_WizardControl != null) {
                        _WizardControl.SelectedPageChanging -= WizardControl_SelectedPageChanging;
                        _WizardControl.SelectedPageChanged -= WizardControl_SelectedPageChanged;
                    }
                    _WizardControl = value;
                    if (!(this.DesignMode)) {
                        CustomizeWizardControl();
                    }
                }
            }
        }

        public TransitionManager TransitionManager {
            get { return _TransitionManager; }
            set {
                if (_TransitionManager != value) {
                    _TransitionManager = value;
                    if (!(this.DesignMode)) {
                        CustomizeTransitionManager();
                    }
                }
            }
        }

        private void CustomizeTransitionType() {
            switch (TransitionType) {
                case TransitionType.PushTransition:
                    PushTransition pt = new PushTransition();

                    transition.TransitionType = pt;
                    break;
                case TransitionType.ClockTransition:
                    transition.TransitionType = new ClockTransition();
                    break;
                case TransitionType.CombTransition:
                    transition.TransitionType = new CombTransition();
                    break;
                case TransitionType.CoverTransition:
                    transition.TransitionType = new CoverTransition();
                    break;
                case TransitionType.DissolveTransition:
                    transition.TransitionType = new DissolveTransition();
                    break;
                case TransitionType.FadeTransition:
                    transition.TransitionType = new FadeTransition();
                    break;
                case TransitionType.ShapeTransition:
                    transition.TransitionType = new ShapeTransition();
                    break;
                case TransitionType.SlideFadeTransition:
                    transition.TransitionType = new SlideFadeTransition();
                    break;
            }
        }

        private void CustomizeTransitionManager() {
            TransitionManager.Transitions.Add(transition);
            TransitionManager.CustomTransition += TransitionManager_CustomTransition;
        }

        void TransitionManager_CustomTransition(ITransition transition, CustomTransitionEventArgs e) {
            int dif = DevExpress.XtraWizard.WizardBaseConsts.CommandAreaHeight;
            e.Regions = new Rectangle[] { new Rectangle(WizardControl.Bounds.X, WizardControl.Bounds.Y, WizardControl.Bounds.Width, WizardControl.Bounds.Height - dif) };
        }

        private void CustomizeWizardControl() {
            WizardControl.AllowTransitionAnimation = false;
            transition.Control = WizardControl;
            WizardControl.SelectedPageChanging += WizardControl_SelectedPageChanging;
            WizardControl.SelectedPageChanged += WizardControl_SelectedPageChanged;
        }

        void WizardControl_SelectedPageChanged(object sender, WizardPageChangedEventArgs e) {
            if (TransitionManager != null) {
                TransitionManager.EndTransition();
            }
        }

        void WizardControl_SelectedPageChanging(object sender, WizardPageChangingEventArgs e) {
            if (TransitionManager != null) {
                TransitionManager.StartTransition(WizardControl);
            }
        }


    }
}

File: VB/WindowsFormsApplication1/WizardControlAnimationHelper.vb
Imports DevExpress.Utils.Animation
Imports DevExpress.XtraWizard
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Namespace WindowsFormsApplication1

    Public Enum TransitionType
        PushTransition
        ShapeTransition
        FadeTransition
        ClockTransition
        DissolveTransition
        SlideFadeTransition
        CoverTransition
        CombTransition
    End Enum

    Public Class WizardControlAnimationHelper
        Inherits Component

        Private transition As Transition
        Private _WizardControl As WizardControl
        Private _TransitionManager As TransitionManager
        Private _TransitionType As TransitionType



        Public Sub New()
            transition = New Transition()
            transition.ShowWaitingIndicator = DevExpress.Utils.DefaultBoolean.False
            TransitionType = WindowsFormsApplication1.TransitionType.PushTransition
        End Sub

        Protected Overrides Sub Finalize()
            If TransitionManager IsNot Nothing Then
                TransitionManager.Transitions.Remove(transition)
            End If
        End Sub

        Public Property TransitionType() As TransitionType
            Get
                Return _TransitionType
            End Get
            Set(ByVal value As TransitionType)
                If _TransitionType <> value Then
                    _TransitionType = value
                    If Not(Me.DesignMode) Then
                        CustomizeTransitionType()
                    End If
                End If
            End Set
        End Property

        Public Property WizardControl() As WizardControl
            Get
                Return _WizardControl
            End Get
            Set(ByVal value As WizardControl)
                If _WizardControl IsNot value Then
                    If _WizardControl IsNot Nothing Then
                        RemoveHandler _WizardControl.SelectedPageChanging, AddressOf WizardControl_SelectedPageChanging
                        RemoveHandler _WizardControl.SelectedPageChanged, AddressOf WizardControl_SelectedPageChanged
                    End If
                    _WizardControl = value
                    If Not(Me.DesignMode) Then
                        CustomizeWizardControl()
                    End If
                End If
            End Set
        End Property

        Public Property TransitionManager() As TransitionManager
            Get
                Return _TransitionManager
            End Get
            Set(ByVal value As TransitionManager)
                If _TransitionManager IsNot value Then
                    _TransitionManager = value
                    If Not(Me.DesignMode) Then
                        CustomizeTransitionManager()
                    End If
                End If
            End Set
        End Property

        Private Sub CustomizeTransitionType()
            Select Case TransitionType
                Case TransitionType.PushTransition
                    Dim pt As New PushTransition()

                    transition.TransitionType = pt
                Case TransitionType.ClockTransition
                    transition.TransitionType = New ClockTransition()
                Case TransitionType.CombTransition
                    transition.TransitionType = New CombTransition()
                Case TransitionType.CoverTransition
                    transition.TransitionType = New CoverTransition()
                Case TransitionType.DissolveTransition
                    transition.TransitionType = New DissolveTransition()
                Case TransitionType.FadeTransition
                    transition.TransitionType = New FadeTransition()
                Case TransitionType.ShapeTransition
                    transition.TransitionType = New ShapeTransition()
                Case TransitionType.SlideFadeTransition
                    transition.TransitionType = New SlideFadeTransition()
            End Select
        End Sub

        Private Sub CustomizeTransitionManager()
            TransitionManager.Transitions.Add(transition)
            AddHandler TransitionManager.CustomTransition, AddressOf TransitionManager_CustomTransition
        End Sub

        Private Sub TransitionManager_CustomTransition(ByVal transition As ITransition, ByVal e As CustomTransitionEventArgs)
            Dim dif As Integer = DevExpress.XtraWizard.WizardBaseConsts.CommandAreaHeight
            e.Regions = New Rectangle() { New Rectangle(WizardControl.Bounds.X, WizardControl.Bounds.Y, WizardControl.Bounds.Width, WizardControl.Bounds.Height - dif) }
        End Sub

        Private Sub CustomizeWizardControl()
            WizardControl.AllowTransitionAnimation = False
            transition.Control = WizardControl
            AddHandler WizardControl.SelectedPageChanging, AddressOf WizardControl_SelectedPageChanging
            AddHandler WizardControl.SelectedPageChanged, AddressOf WizardControl_SelectedPageChanged
        End Sub

        Private Sub WizardControl_SelectedPageChanged(ByVal sender As Object, ByVal e As WizardPageChangedEventArgs)
            If TransitionManager IsNot Nothing Then
                TransitionManager.EndTransition()
            End If
        End Sub

        Private Sub WizardControl_SelectedPageChanging(ByVal sender As Object, ByVal e As WizardPageChangingEventArgs)
            If TransitionManager IsNot Nothing Then
                TransitionManager.StartTransition(WizardControl)
            End If
        End Sub


    End Class
End Namespace
